### PR TITLE
[Issue #1449] Update removeJob command to refresh the profiles cache separately from refreshing a tree

### PR DESCRIPTION
**Pull request by nickImbirev**
## Proposed changes

Resolves the issue #1449 by changing semantics of the `refreshAll` util function. Now, we need to refresh the profiles cache independently from any tree provider.

https://user-images.githubusercontent.com/74672982/136568136-6b05769d-40a1-4902-8068-a3c610e781cf.mov

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [x] I have added necessary documentation (if appropriate)
- [x] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by jellypuno — DISMISSED** [review]
lgtm!

**Comment by JillieBeanSim** [comment]
@jellypuno @nickImbirev this is what I am seeing testing this.  Errors pop up if deleted when opened by clicking the job id link plus what I mentioned above.  I will also test against main to see if any of it is pre-existing (I don't remember seeing this issue) or if it's a regression
job-deletion

File: packages/zowe-explorer/src/job/actions.ts [hunk 1]
```diff
@@ -248,63 +249,110 @@ export async function setPrefix(job: IZoweJobTreeNode, jobsProvider: IZoweTree<I
...
+    }
+}
 
-    if (nodes.length > 0) {
-        for (const node of nodes) {
-            nodesToDelete.push(`${node.job.jobname}(${node.job.jobid})`);
-        }
-    } else if (job) {
-        nodesToDelete.push(`${job.job.jobname}(${job.job.jobid})`);
+const deleteSingleJob = async (job: IZoweJobTreeNode, jobsProvider: IZoweTree<IZoweJobTreeNode>): Promise<void> => {
```
Comment (jellypuno) [inline packages/zowe-explorer/src/job/actions.ts:270]: Can you please change this into our standard format? something like this `export async function deleteSingleJob`. At least i want our functions to be coded in the same way. this is supposed to be a function, right?

**Review by jellypuno — CHANGES_REQUESTED** [review]

File: packages/zowe-explorer/src/job/actions.ts (hunk 1 above)
Comment (nickImbirev) [reply packages/zowe-explorer/src/job/actions.ts]: Yeah, sure, I can do that, thank you for bringing this up!

**Review by jellypuno — APPROVED** [review]
Looks good to me! Thanks @nickImbirev

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @nickImbirev for the fix

I do see some broken integration tests but also on Master branch.  Created issue for that
https://github.com/zowe/vscode-extension-for-zowe/issues/1595

I also see some files show up in Source Control that deal with the new FTP unit tests that should be ignored.  Created issue for that too
https://github.com/zowe/vscode-extension-for-zowe/issues/1594

**Review by jellypuno — APPROVED** [review]

**Pull request merged**